tests/helper.py

```diff
@@ -22,6 +25,11 @@ def assert_dtype_allclose(dpnp_arr, numpy_arr, check_type=True):
             numpy.finfo(numpy_arr.dtype).resolution,
         )
         assert_allclose(dpnp_arr.asnumpy(), numpy_arr, atol=tol, rtol=tol)
+        if check_type:
+            if has_support_aspect64(dpnp_arr.sycl_device):
```

Comment:
It seems you also need to add `has_support_aspect16`.
```suggestion
            if has_support_aspect64(dpnp_arr.sycl_device) and has_support_aspect16(dpnp_arr.sycl_device):
```

Reply:
@vtavana Yes, adding `has_support_aspect16` check is a good point, thanks!
But your suggestion will always return False on cpu so I added a bit more complicated check for that